### ISSUE TITLE
LDAPI (LDAP over IPC) exception in ldap3/strategy/base.py

### ISSUE THREAD

**Issue by golebiewsky** [cannatag/ldap3#837]
Hi,

I'm trying to use LDAPI connection on my server [using this tutorial](https://ldap3.readthedocs.io/en/latest/bind.html?highlight=ldapi#ldapi-ldap-over-ipc).
```
Python 3.8.3 (default, Jun 30 2020, 17:25:00)
[GCC 8.3.1 20191121 (Red Hat 8.3.1-5)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from ldap3 import Server, Connection, SASL, EXTERNAL
>>> s = Server('ldapi:///var/run/slapd-REDACTED-COM.socket')
>>> c = Connection(s, authentication=SASL, sasl_mechanism=EXTERNAL, sasl_credentials='')
>>> c.bind()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/pass_reminder/venv3.8/lib/python3.8/site-packages/ldap3/core/connection.py", line 561, in bind
    self.open(read_server_info=False)
  File "/opt/pass_reminder/venv3.8/lib/python3.8/site-packages/ldap3/strategy/sync.py", line 56, in open
    BaseStrategy.open(self, reset_usage, read_server_info)
  File "/opt/pass_reminder/venv3.8/lib/python3.8/site-packages/ldap3/strategy/base.py", line 139, in open
    raise exception_history[0][0]
ldap3.core.exceptions.LDAPSocketOpenError: a bytes-like object is required, not 'tuple'
>>>
```
Above exception is rasied. I tried also with server name with slashes to path changed like this
`'ldapi://%2fvar%2frun%2fslapd-INTRA-REDACTED-COM.socket'` but I got the same exception.
The socket file exists.

My enviroment:
OS: CentOS Linux release 8.2.2004 (Core)
Python version: 3.8.3
LDAP server: FreeIPA, version: 4.8.4

**Comment by cannatag** [comment]
same as #803

**Comment by cannatag** [comment]
Hi, can you try the code in the dev branch? LDAPI should work again.

Thanks,
Giovanni

**Comment by cannatag** [comment]
Fixed in ldap3 v2.8.
